### PR TITLE
Implement a global gProfiler lock

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Implement a global gProfiler lock, to ensure only 1 gProfiler is run at a time.

## Motivation and Context
gProfiler is a system-wide profiler, so no more than 1 instance should run concurrently.

Besides the unnecessary performance "cost" of running 2 simultaneously, there's also a certain risk with profilers that are not fully external in their operation: for example, async-profiler is loaded into target processes and gProfiler manages it by sending commands with `jattach`. 2 gProfilers trying to profile the same Java process could potentially get their commands mixed, resulting, for example, in async-profiler kept in profiling due to a missed `stop` command (this itself should be fixed in the `java.py` profiler logic - it should handle such unexpected cases on itself and abort profiling gracefully; but we should add this additional layer of protection with the global lock).

## How Has This Been Tested?
Tested locally - ran gProfiler in a container (w/o `--network=host`) and then tried to run another one:
* container with `--network=host`
* container without `--network=host`
* standalone executable

All refused to execute.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the relevant documentation.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: gprofiler/utils.py [hunk 1]
```diff
@@ -261,6 +266,30 @@ def get_run_mode() -> str:
         return "local_python"
 
 
+# we can't setns(CLONE_NEWNS) in a multithreaded program (unless we unshare(CLONE_NEWNS) before)
+# so, we start a new thread, unshare() & setns() it, get our needed information and then stop this thread
+# (so we don't keep unshared threads running around)
+# for other namespace types, we use this function to execute callbacks without changing the namespaces for the
+# core threads.
+def run_in_init_ns(nstype: str, callaback: Callable[[], None]) -> None:
```
Comment (guybortnikov) [inline gprofiler/utils.py:274]:
```suggestion
def run_in_init_ns(nstype: str, callback: Callable[[], None]) -> None:
```

**Review by guybortnikov — CHANGES_REQUESTED** [review]
Looks good, small typo fix and can be merged.

**Review by guybortnikov — APPROVED** [review]

File: gprofiler/main.py [hunk 2]
```diff
@@ -27,7 +27,15 @@
 from .java import JavaProfiler
 from .perf import SystemProfiler
 from .python import get_python_profiler
-from .utils import is_root, run_process, get_iso8061_format_time, resource_path, log_system_info, TEMPORARY_STORAGE_PATH
+from .utils import (
```
Comment (talsaiag) [inline gprofiler/main.py:30]: @Jongy wtf is this formatting?

Comment (Jongy) [reply gprofiler/main.py:30]: Default `black` settings?

Comment (talsaiag) [reply gprofiler/main.py:30]: lol looks ugly (but effective git diff wise, since each is at its own line)

Comment (Jongy) [reply gprofiler/main.py:30]: I don't really mind how it looks tbh, as long as it just works (and yeah, the git diff is nice)

**Pull request merged**